### DISCUSSION TITLE
允許雙向翻譯

### DISCUSSION THREAD

**Discussion by nissenyeh** [immersive-translate/immersive-translate#1071] [Ideas]
### 新功能 | New Feature

1. 允許雙向翻譯：

舉例來說
- 當偵測為英文時，可以自動將「目標語言」設定為中文
- 但是當偵測為中文時，可以自動將「目標語言」設定為英文

### 为什么您想要添加这些功能 | Why you want to add these features

目前只能設置單向翻譯，如果要修改，會需要手動更改設置，但希望可以允許設定雙向翻譯

**Comment by theowenyoung** [comment]
这个实现倒是很简单，但是考虑到增加这个设置会显著提高用户的理解成本，所以我们应该不会添加这个功能。

替代方案是使用2个浏览器，分别使用不同的设置。

**Comment by nissenyeh** [comment]
理解，謝謝回覆！

這邊還是提供一個可能的設計（ 覺得 google 翻譯的雙向箭頭算是蠻直覺的，但單純供參考）

![CleanShot 2023-11-03 at 02 16 27](https://github.com/immersive-translate/immersive-translate/assets/34571901/fc874723-32bd-43c1-a0a3-4e13700c81cc)

![CleanShot 2023-11-03 at 02 25 04](https://github.com/immersive-translate/immersive-translate/assets/34571901/042f5751-3451-4606-9486-c9baef3de819)

另外這個插件真的幫助到我很多，感謝你們

**Reply by theowenyoung** [reply]
<img width="1364" alt="image" src="https://github.com/immersive-translate/immersive-translate/assets/62473795/8745dedb-bc5f-4d2f-b31b-0d62675257b8">

我测试发现，谷歌的页面似乎并不能自动切换目标语言？


还有一点就是 沉浸式翻译 目前的主要目的是辅助用户阅读，而不是学习，所以有意缺失很多学习相关的功能，比如 划词翻译，生词本等等等等。

我可能觉得 将中文反向翻译为英文 确实是很小众的功能，加上之后还是会让很多用户觉得困惑。（目前的门槛已经很高了.. 我们有一半的用户安装之后不会用...

**Reply by nissenyeh** [reply]
對，不能自動切換，只是一個雙向箭頭的示意圖而已

完全理解！確實還是不要太複雜的好

感謝
